### PR TITLE
Remove isBuiltIn and builtinModules

### PR DESCRIPTION
**Pull request by vicb**
Following [this discussion](https://github.com/cloudflare/workerd/discussions/2642) this PR removes `.isBuiltin` and `.builtinModules`.

/cc @jasnell @pi0 @petebacondarwin @IgorMinar

### CODE REVIEW

**Comment by vicb** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by vicb** [comment]
Not sure how to format the code, it would be a nice addition to CONTRIBUTING.md

**Comment by petebacondarwin** [comment]
> Not sure how to format the code, it would be a nice addition to CONTRIBUTING.md

Looks you just need to run

```bash
python3 ./tools/cross/format.py
```

In the CI it runs this with `--check`.

File: src/node/internal/module.d.ts [hunk 1]
```diff
@@ -2,5 +2,6 @@
 // Licensed under the Apache 2.0 license found in the LICENSE file or at:
 //     https://opensource.org/licenses/Apache-2.0
 
-export function createRequire(path: string): (specifier: string) => unknown;
-export function isBuiltin(specifier: string): boolean;
+export function createRequire(
+  path: string | URL
```
Comment (anonrig) [inline src/node/internal/module.d.ts:6]: This should be path: string. The previous code was correct.

Comment (vicb) [reply src/node/internal/module.d.ts:6]: oh right that's the internal module, I'll revert, thanks for catching!

**Comment by vicb** [comment]
Can we get the next?

**Comment by jasnell** [comment]
Closing as these have already gone out into a release.

**Pull request closed without merging**